File: abs/golang/arc065_a.go
package main

import "fmt"

func main() {
	var (
		s string
	)
	fmt.Scan(&s)
	stock := []string{"dream", "dreamer", "erase", "eraser"}

	for len(s) > 0 {
		found := false
		for k := 0; k < len(stock); k++ {
			tmp := stock[k]
			pos := len(s) - len(tmp)
			if pos >= 0 && s[pos:] == tmp {
				s = s[:pos]
				found = true
			}
		}
		if !found {
			break
		}
	}

	if len(s) == 0 {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}

}

File: abs/rust/arc065_a.rs
// https://qiita.com/tubo28/items/e6076e9040da57368845

use std::io::*;
use std::str::FromStr;


fn read<T: FromStr>() -> T {
    let stdin = stdin();
    let stdin = stdin.lock();
    let token: String = stdin
        .bytes()
        .map(|c| c.expect("failed to read char") as char)
        .skip_while(|c| c.is_whitespace())
        .take_while(|c| !c.is_whitespace())
        .collect();
    token.parse().ok().expect("failed to parse token")
}

fn main () {
    let ss: String = read();
    let mut s = &ss[..];
    let words = ["dream", "dreamer", "erase", "eraser"];

    let mut found = false;
    while !s.is_empty() {
        found = false;
        for w in &words {
            if s.ends_with(w) {
                s = &s[0..s.len() - w.len()];
                found = true;
            }
        }
        if !found {
            break;
        }
    }

    let ans = if found {"YES"} else {"NO"};

    println!("{}", ans);
}